cqrs/src/main/scala/it/pagopa/interop/commons/cqrs/model/PartialMongoAction.scala

```diff
@@ -0,0 +1,46 @@
+package it.pagopa.interop.commons.cqrs.model
+
+import org.mongodb.scala.{Document, SingleObservable}
+import org.mongodb.scala.bson.conversions.Bson
+
+/**
+  * The partial action is used to intercept and enhance parameters passed to the mongo command.
+  * Example: Adding metadata to documents saved and updated
+  */
+trait PartialMongoAction
+
+/**
+  * Used when the value parameter is of type Bson, for example when updating using functions like Updates.set or Updates.pull
+  * 
+  * Usage:
+  * {{{
+  * ActionWithBson(
+  *   value => collection.updateOne(Filters.eq("data.id", "xyz"), value), 
+  *   Updates.set("data", "data")
+  * )  
+  * }}}
```

Comment:
Curios doc style, were you thinking to add something like `https://github.com/tkawachi/sbt-doctest`? It might be worth for commons :)

Reply:
Not really, it was just for code block formatting 😄